Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -126,6 +126,12 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
                     return buffer.validate().bind { validationResult in
                         if let invalidError = validationResult.failureValue as? BufferInvalidError {
                             self.statsSession.validationStats = self.validationStatsFrom(error: invalidError)
+
+                            log.warning("Buffer inconsistent, starting repair procedure")
+                            let repairer = BookmarksRepairRequestor(scratchpad: self.scratchpad, basePrefs: self.basePrefs, remoteClients: remoteClientsAndTabs)
+                            return repairer.startRepairs(validationInfo: invalidError.inconsistencies) >>> {
+                                deferMaybe(invalidError)
+                            }
                         }
                         return deferMaybe(result)
```

Comment:
It's time to invert the conditional into a `guard`, so that this isn't buried down here.